eureka/lib/simultaneous_order_fitting.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import scipy.optimize as so
+import pyximport
+import itertools
+pyximport.install()
+from ..S3_data_reduction import niriss_extraction as profiles
+
+__all__ = ['fit_orders_fast', 'fit_orders']
+
+
+def fit_orders(data, tab):
+    """
+    Creates a 2D image optimized to fit the data. Currently
+    runs with a Gaussian profile, but will look into other
+    more realistic profiles at some point. This routine
+    is a bit slow, but fortunately, you only need to run it
+    once per observations.
+
+    Parameters
+    ----------
+    data : object
+    which_table : int, optional
+       Sets with table of initial y-positions for the
+       orders to use. Default is 2.
+
+    Returns
+    -------
+    ord1 : np.ndarray
+    ord2 : np.ndarray
+    """
+    print("Go grab some food. This routing could take up to 30 minutes.")
+
+    def construct_guesses(A, B, sig, length=10):
+        # amplitude of gaussian for first order
+        As = np.linspace(A[0], A[1], length)
+        # amplitude of gaussian for second order
+        Bs = np.linspace(B[0], B[1], length)
+        # std of gaussian profile
+        sigs = np.linspace(sig[0], sig[1], length)
+        # generates all possible combos
+        combos = np.array(list(itertools.product(*[As, Bs, sigs])))
```

Comment:
Why convert to a list and then convert to an array? Also, why do `*[As,Bs,sigs]` instead of just `As, Bs, sigs`?